control-plane/connect-inject/peering_acceptor_controller.go

```diff
@@ -0,0 +1,335 @@
+package connectinject
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+	consulv1alpha1 "github.com/hashicorp/consul-k8s/control-plane/api/v1alpha1"
+	"github.com/hashicorp/consul/api"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// PeeringAcceptorController reconciles a PeeringAcceptor object.
+type PeeringAcceptorController struct {
+	client.Client
+	// ConsulClient points at the agent local to the connect-inject deployment pod.
+	ConsulClient *api.Client
+	Log          logr.Logger
+	Scheme       *runtime.Scheme
+	context.Context
+}
+
+//+kubebuilder:rbac:groups=consul.hashicorp.com,resources=peeringacceptors,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=consul.hashicorp.com,resources=peeringacceptors/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=apps,resources=secrets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=apps,resources=secrets/status,verbs=get
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// PeeringAcceptor resources determine whether to generate a new peering token in Consul and store it in the backend
+// specified in the spec. If the resource doesn't exist, the peering should be deleted in Consul. If the resource
+// exists, and a peering doesn't exist in Consul, it should be created. If the resource exists, and a peering does exist
+// in Consul, it should be reconciled. If the status of the resource does not match the current state of the specified
+// secret, generate a new token and store it according to the spec.
```

Comment:
Might be easier to parse as a list? 🤷🏻 

```suggestion
// PeeringAcceptor resources determine whether to generate a new peering token in Consul and store it in the backend
// specified in the spec. 
//  - If the resource doesn't exist, the peering should be deleted in Consul. 
//  - If the resource exists and a peering doesn't exist in Consul, it should be created. 
//  - If the resource exists and a peering does exist in Consul, it should be reconciled. 
//  - If the status of the resource does not match the current state of the specified
//    secret, generate a new token and store it according to the spec.
```

Reply:
Yup! done!